chain/network/src/routing/routing_table_view.rs

```diff
@@ -88,31 +89,33 @@ impl RoutingTableView {
     /// Find peer that is connected to `source` and belong to the shortest path
     /// from `source` to `peer_id`.
     pub fn find_route_from_peer_id(&mut self, peer_id: &PeerId) -> Result<PeerId, FindRouteError> {
-        if let Some(routes) = self.peer_forwarding.get(peer_id).cloned() {
-            if routes.is_empty() {
-                return Err(FindRouteError::Disconnected);
+        if let Some(routes) = self.peer_forwarding.get(peer_id) {
+            match (routes.iter())
+                .map(|peer_id| {
+                    (self.route_nonce.get(peer_id).cloned().unwrap_or_default(), peer_id)
+                })
+                .minmax()
+                .into_option()
+            {
+                None => Err(FindRouteError::Disconnected),
+                // Neighbor with minimum and maximum nonce respectively.
+                Some(((min_v, next_hop), (max_v, _))) => {
+                    // Strategy similar to Round Robin. Select node with least nonce and send it. Increase its
+                    // nonce by one. Additionally if the difference between the highest nonce and the lowest
+                    // nonce is greater than some threshold increase the lowest nonce to be at least
+                    // max nonce - threshold.
+                    self.route_nonce.put(
+                        next_hop.clone(),
+                        // TODO, should be max(min_v + 1, max_v.saturating_sub(ROUND_ROBIN_MAX_NONCE_DIFFERENCE_ALLOWED))
```

Comment:
does it matter? or should we just update the comment in line 106 and leave it as is?

Reply:
@mm-near It doesn't matter. I removed the comment.